engine/crates/runtime/src/trusted_documents.rs

```diff
@@ -0,0 +1,33 @@
+#[derive(Debug)]
+pub enum TrustedDocumentsError {
+    RetrievalError(Box<dyn std::error::Error + Send + Sync + 'static>),
+    DocumentNotFound,
+}
+
+pub type TrustedDocumentsResult<T> = Result<T, TrustedDocumentsError>;
+
+/// A handle to trusted documents configuration and retrieval.
+pub struct TrustedDocuments {
+    inner: Box<dyn TrustedDocumentsImpl>,
+    branch_id: String,
+}
+
+impl TrustedDocuments {
+    pub fn new(inner: Box<dyn TrustedDocumentsImpl>, branch_id: String) -> Self {
```

Comment:
Here we'll need to take the headers you can configure to bypass trusted documents enforcement.

Reply:
And maybe move more of the enforcement logic here, for simplicity.